lsm-tree.cabal

```diff
@@ -130,10 +130,10 @@ library
     Database.LSMTree.Internal.CRC32C
     Database.LSMTree.Internal.Cursor
     Database.LSMTree.Internal.Entry
-    Database.LSMTree.Internal.IndexCompact
-    Database.LSMTree.Internal.IndexCompactAcc
-    Database.LSMTree.Internal.IndexOrdinary
-    Database.LSMTree.Internal.IndexOrdinaryAcc
+    Database.LSMTree.Internal.Index.Compact
+    Database.LSMTree.Internal.Index.CompactAcc
+    Database.LSMTree.Internal.Index.Ordinary
+    Database.LSMTree.Internal.Index.OrdinaryAcc
```

Comment:
I think that expressing inherent structure is generally a good idea, which is why I introduced this additional nesting of module names. Of course, this should be weighed against difficulties this might cause.

As far as the source code itself is concerned, the only consequence of this change is that a dot has to be put here and there. Regarding the organization of the source code in the file system, there is the need for an additional subdirectory, which may or may not be considered burdensome. I myself tend to refer to individual source files via patterns like `src/**/Compact.hs`, which are supported by Vim, by Bash with the `globstar` option, and by the Z Shell (where they come from, as far as I know). With this approach, things even become simpler with this deeper hierarchy; for example, I don’t have to write the longer `src/**/IndexCompact.hs` anymore. The situation may be the other way round with different tools and different tool usage.

All in all, I prefer to have this change. You and @mheinzel said outside of GitHub that you don’t have a strong bias towards one or the other solution. I’d say that, if @dcoutts doesn’t have any strong opinion on this, we should go for this change.

Reply:
> Does `src/**/*Compact.hs` work in your case?

Yes, it does, in both Vim and Bash (with `globstar`). I don’t have the Z Shell installed, but I’m pretty sure that it works in it as well.

Reply:
I think it's okay to assume we can include the reorganisation.